[MN] more cleanup over the tier two area.

furszy opened the pull request:
This time removing:

1) `GetOldestBlock` & `GetNewestBlock` unused methods from CMasternodePayments. 
2) `pubKeyCollateralAddress1` & `pubKeyMasternode1` unused members from CMasternode.
3) Direct `chainActive` access in `GetLastPaid` from CMasternode.

Plus:
1) Added a missing cs lock in `CMasternode::Check`

random-zebra reviewed (dismissed):
utACK 7e1354a7139f2410b0a9c5be28dd3509f9528c47

furszy commented:
rebased on top of master and replaced the `CMasternode::Check()` lock for a to do. We need to be careful there, `cs_main` is being locked inside the function as well.

This ca be merged as is now.

random-zebra reviewed (approved):
re-utACK c587c7364eeb5e468d65e2679d9ebd7f1deb9229 and merging...
